ili/utils/samplers.py

```diff
@@ -74,25 +74,33 @@ def sample(self, nsteps: int, x: np.ndarray,
             progress (bool, optional): whether to show progress bar.
                 Defaults to False.
         """
-        theta0 = np.stack([self.posterior.prior.sample()
-                          for i in range(self.num_chains)])
+        theta0 = [self.posterior.prior.sample()
+                          for i in range(self.num_chains)]
+        if not isinstance(theta0[0], np.ndarray):
+            theta0 = [t.cpu() for t in theta0]
+        theta0 = np.stack(theta0)
 
         def log_target(t, x):
-            return np.array(self.posterior.potential(
+            res = self.posterior.potential(
                 t.astype(np.float32), x.astype(np.float32)
-            ))
+            )
+            if hasattr(res, 'cpu'):
+                res = np.array(res.cpu())
+            return res
+
         self.sampler = emcee.EnsembleSampler(
             self.num_chains,
             theta0.shape[-1],
             log_target,
             vectorize=False,
-            args=(x,)
+            args=(x,),
         )
         self.sampler.run_mcmc(
             theta0,
             self.burn_in + nsteps,
             thin_by=self.thin,
             progress=progress,
+            **kwargs
```

Comment:
Are these **kwargs necessary? From [the emcee sampler interface](https://emcee.readthedocs.io/en/stable/user/sampler/), it doesn't seem like there's many additional arguments that we would need to specify.

Reply:
Indeed we were only using the option `skip_initial_state_check`, so I have removed `**kwargs` and replaced it with that input.